org.eclipse.scanning.device.ui/src/org/eclipse/scanning/device/ui/Activator.java

```diff
@@ -79,5 +88,22 @@ public ImageData getImageData() {
 		return imageDescriptorFromPlugin(PLUGIN_ID, path);
 	}
 
+	@Override
+	public <T> T getService(Class<T> serviceClass) {
+		if (context==null) return null;
+		ServiceReference<T> ref = context.getServiceReference(serviceClass);
+		return context.getService(ref);
+	}
+
+	@Override
+	public <T> Collection<T> getServices(Class<T> serviceClass) throws InvalidSyntaxException {
+		if (context==null) return null;

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Return an empty collection instead of null. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1168)